File: src/y.clj
(println
 (do

   ;; Fixpoint

   (def error (fn [n] 'error))

   (def y
        (fn [ff]
          ((fn [g] (fn [n] ((ff (g g)) n)))
           (fn [g] (fn [n] ((ff (g g)) n))))))

   (def fact-partial
        (fn [recurse]
          (fn [n]
            (if (zero? n) 1 (* n (recurse (- n 1)))))))

   (def fact (y fact-partial))

   (fact 5)

   ))

File: src/y.rb
p ->() {

  fp = ->(fact) {
    ->(n) { n==0 ? 1 : n*fact.(n-1) }
  }

  y = ->(fun) {
    ->(g) { ->(n) { fun.(g.(g)).(n) } }.(
      ->(g) { ->(n) { fun.(g.(g)).(n) } })
  }

  factorial = y.(fp)
  factorial.(5)

  #  y.(fp) == fp(factorial) == factorial
  #
  # y calculates the fixpoint of a partially recursive function

}.()
